### ISSUE TITLE
`reportUnsafeMultipleInheritance` should not complain on dataclasses

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#341]
```py
from dataclasses import dataclass


@dataclass
class Foo:
    a: int


@dataclass
class Bar:
    b: str


@dataclass
class Baz(Foo, Bar): ...  # error: reportUnsafeMultipleInheritance
```

as far as i can tell this is completely safe, so there should be no error unless it uses `init=False`
